### PR TITLE
CDRIVER-3937 Remove usages of xtrace in build scripts

### PR DESCRIPTION
**Pull request by alcaeus**
CDRIVER-3937

The first commit is necessary as the `remote_file` is prepended with the project dir automatically. The already generated config.yml was correct, but regenerating broke it without this commit.

### CODE REVIEW

File: build/evergreen_config_lib/variants.py [hunk 1]
```diff
@@ -635,7 +635,7 @@ def days(n):
         'ubuntu1604-small',
         ['.tsan'],
         {'CC': '/opt/mongodbtoolchain/v3/bin/clang'},
-        batchtime=days(1))
+        batchtime=days(1)),
```
Comment (benjirewis) [inline build/evergreen_config_lib/variants.py:638]: Ah thank you!

**Review by benjirewis — APPROVED** [review]
LGTM!

File: build/evergreen_config_lib/tasks.py [hunk 2]
```diff
@@ -390,7 +390,7 @@ def __init__(self, task_name, suffix_commands, orchestration=True, **kwargs):
                          remote_file='${branch_name}/mongo-c-driver-debian-packages-${CURRENT_VERSION}.tar.gz',
                          content_type='${content_type|application/x-gzip}'),
                   s3_put(local_file='deb.tar.gz',
-                         remote_file='${project}/${branch_name}/${revision}/${version_id}/${build_id}/${execution}/mongo-c-driver-debian-packages.tar.gz',
+                         remote_file='${branch_name}/${revision}/${version_id}/${build_id}/${execution}/mongo-c-driver-debian-packages.tar.gz',
                          content_type='${content_type|application/x-gzip}')]),
```
Comment (rcsanchez97) [inline build/evergreen_config_lib/tasks.py:394]: This change seems unrelated to the xtrace issue.  Also, this change would make the C driver publish distro package artifacts under a path that is not consistent with the pattern used in libmongocrypt and the C++ driver.  Can `${project}/` be restored?

File: build/evergreen_config_lib/tasks.py [hunk 3]
```diff
@@ -399,7 +399,7 @@ def __init__(self, task_name, suffix_commands, orchestration=True, **kwargs):
                          remote_file='${branch_name}/mongo-c-driver-rpm-packages-${CURRENT_VERSION}.tar.gz',
                          content_type='${content_type|application/x-gzip}'),
                   s3_put(local_file='rpm.tar.gz',
-                         remote_file='${project}/${branch_name}/${revision}/${version_id}/${build_id}/${execution}/mongo-c-driver-rpm-packages.tar.gz',
+                         remote_file='${branch_name}/${revision}/${version_id}/${build_id}/${execution}/mongo-c-driver-rpm-packages.tar.gz',
```
Comment (rcsanchez97) [inline build/evergreen_config_lib/tasks.py:402]: Ibid.

File: build/evergreen_config_lib/tasks.py (hunk 2 above)
Comment (alcaeus) [reply build/evergreen_config_lib/tasks.py:394]: This change was introduced in https://github.com/mongodb/mongo-c-driver/commit/c883dab9ff915b2cff60f2ad0ac6695094506eed, but regenerating the config file introduces a double `${project}/` path component in config.yml. This is confirmed by the surrounding lines, where the `remote_file` option does not start with `${project}`, and by the fact that while the template in tasks.py changes, the generated config file does not.

Comment (rcsanchez97) [reply build/evergreen_config_lib/tasks.py:394]: OK.  That was totally not expected.  It is clearly a copy/paste on my part in the commit where I introduced the change.  Thanks for catching and fixing it and thanks for the explanation!

**Review by rcsanchez97 — APPROVED** [review]
LGTM
